### DISCUSSION TITLE
Vaultwarden admin password

### DISCUSSION THREAD

**Discussion by andyw73** [tteck/Proxmox#1661] [General] [answered]
I have run the script bash -c "$(wget -qLO - https://github.com/tteck/Proxmox/raw/main/ct/vaultwarden.sh)" and set the vaultwarden admin token. This is seen in /opt/vaultwarden/.env and is converted to ADMIN_TOKEN='$argon2i$v=#######'. I am unable to login to the admin panel with set password and receive a Error: Invalid admin token, please try again.
Has anyone else run into this issue?

**Comment by tteck** [comment]
Please utilize the password you entered during the "Set the ADMIN_TOKEN" process.

**Reply by andyw73** [reply]
I have done that however it still gives an Error: Invalid admin token, please try again.
When changing the password the argon value does change so I know the proccess is working however I am still unable to access admin page saying the password is incorrect.

**Reply by chaot2000** [reply]
Same here. I used the script to update - all are fine.

Changing Admin Token - always the same error message "Error: Invalid admin token, please try again."

**Reply by tteck** [reply]
Upon retesting, I entered "password" on the "Set ADMIN_TOKEN" screen. I then proceeded to IP:8000/admin and entered "password" without encountering any issues. (tested 3 times, changing password each time)

**Reply by chaot2000** [reply]
I've got a brand new Installation from Vaultwarden. This solved the ADMIN_TOKEN problem.

But, i can't import the password from my old instance. Error comes up with 

"Cannot read properties of null (reading '0')"

it's really annoying

**Reply by Node815** [reply]
Did you export from CSV by chance?  If so, try using the JSON export method if you can.  I use the DOCKER version of Vaultwarden (testing release) and had to rebuild my server a few weeks ago, and it failed to restore from the CSV to the same version, but JSON worked perfectly fine.  Also, Vaultwarden just released 1.3.0 today to stable branches which I think now supports the much awaited Passkey support.

**Reply by chaot2000** [reply]
@PDXUser thanks for the hint.

After reinstalling with the ingenious script, I was unable to import a .JSON. With CSV it worked afterwards. 
I have now successfully moved everything!

Many thanks for your support! 
Closed from my site

**Comment by daschmidt1994** [comment]
is there already a solution?
Have the same problem if i try to change the password

**Reply by andyw73** [reply]
In my case there was an extra admin token located in /opt/vaultwarden/data/config.json. I commented out the "admin_token" using nano and re run the script. I was then able to login to the admin panel again. Hope this helps anyone else facing this issue.

**Reply by navels** [reply]
> In my case there was an extra admin token located in /opt/vaultwarden/data/config.json. I commented out the "admin_token" using nano and re run the script. I was then able to login to the admin panel again. Hope this helps anyone else facing this issue.

Thanks, same.

**Comment by navels** [comment]
Got a repro for this problem:
1. Delete the `admin_token` from `/opt/vaultwarden/data/config.json`
3. Set the admin token using the `update` script, which writes it to `/opt/vaultwarden/.env`.
4. Log into the admin panel using the new token
5. Click Save in the admin panel

This causes the `admin_token` to be saved back to /top/vaultwarden/data/config.json`.  At this point the same token is in both files.

6. Change the admin token with the `update` script.

At this point the token in the two files are different, with the older token taking precedence when logging into the admin panel.

**Comment by navels** [comment]
@tteck fyi discussing this wonky (imo) behavior of vaultwarden's over here: https://github.com/dani-garcia/vaultwarden/discussions/4758

**Reply by tteck** [reply]
I don't have this issue.?

**Reply by navels** [reply]
> I don't have this issue.?

Are you following the steps:

1. Delete token from `data/config.json`
2. Set token to `foo` (using `update`)
3. Log into admin panel
4. Click save
5. Change token to `bar` (using `update`)
6. Log into admin panel with `bar` . . . won't work

**Reply by tteck** [reply]
I just created a new Vaultwarden LXC, typed `update` in the console, selected `3  Set Admin Token` and Set the ADMIN_TOKEN (used password). Then in a new browser tab went to `http://192.168.86.250:8000/admin` entered `password` and no issues.

**Reply by tteck** [reply]
I just changed the Admin Token, still no issues.

**Reply by navels** [reply]
Well I'm baffled if you are really following those steps exactly, in order, including clicking the Save button in the admin panel.

**Reply by tteck** [reply]
What I have discovered, if you click save, you're stuck using the original token.

**Reply by navels** [reply]
Egggggggsactly :-)

**Reply by tteck** [reply]
I've found if you change your token after clicking save, you have to click Reset defaults, logout and use your new token to gain access. Long story short, set the token you want before clicking save.

**Reply by navels** [reply]
I'm waiting to see what the vaultwarden folks say but I'd be inclined to have the update script set the token in both places (`.env` and `data/config.json`).  I can submit a PR if you like.

**Reply by andyw73** [reply]
The reason most people would want to change their password is due to being lost or forgoten. I agree with Navels the script should check the existence of both locations and change both locations accordingly. It seems counter intuitive that the password will only be changed in one location as, most will have been in the admin panel to change various parameters according to there setup.

**Reply by navels** [reply]
@andyw73 thanks and if you want to add this to the vaultwarden discussion: https://github.com/dani-garcia/vaultwarden/discussions/4758

**Reply by tteck** [reply, marked as answer]
commit https://github.com/tteck/Proxmox/commit/cfa1b2e04d9056fe96981677bd856f7111ad9495 fixes issue

**Reply by navels** [reply]
Confirmed that fixes it, thanks!
